modules/RTC/TraceablePeerConnection.js

```diff
@@ -689,12 +1088,294 @@ TraceablePeerConnection.prototype.addStream = function(stream, ssrcInfo) {
-    this.trace('removeStream', stream.id);
+/**
+ * Adds local track as part of the unmute operation.
+ * @param {JitsiLocalTrack} track the track to be added as part of the unmute
+ * operation
+ * @return {boolean} <tt>true</tt> if the state of underlying PC has changed and
+ * the renegotiation is required or <tt>false</tt> otherwise.
+ */
+TraceablePeerConnection.prototype.addTrackUnmute = function(track) {
+    if (!this._assertTrackBelongs('addTrackUnmute', track)) {
+        // Abort
+        return false;
+    }
+
+    if (track._isAttachedToPC(this)) {
+        logger.info(`Adding ${track} as unmute to ${this}`);
+        const webRtcStream = track.getOriginalStream();
+
+        if (!webRtcStream) {
+            logger.error(
+                `Unable to add ${track} as unmute to ${this}`
+                    + ' - no WebRTC stream');
 
-    // FF doesn't support this yet.
-    if (this.peerconnection.removeStream) {
-        this.peerconnection.removeStream(stream);
+            return false;
+        }
+        this._addStream(webRtcStream);
+
+        return true;
+    }
+
+    logger.info(`Not adding detached ${track} as unmute to ${this}`);
+
+    return false;
+};
+
+/**
+ * Adds WebRTC media stream to the underlying PeerConnection
+ * @param {MediaStream} mediaStream
+ * @private
+ */
+TraceablePeerConnection.prototype._addStream = function(mediaStream) {
+    this.peerconnection.addStream(mediaStream);
+};
+
+/**
+ * Removes WebRTC media stream from the underlying PeerConection
+ * @param {MediaStream} mediaStream
+ */
+TraceablePeerConnection.prototype._removeStream = function(mediaStream) {
+    if (RTCBrowserType.isFirefox()) {
+        this._handleFirefoxRemoveStream(mediaStream);
+    } else {
+        this.peerconnection.removeStream(mediaStream);
+    }
+};
+
+/**
+ * This method when called will check if given <tt>localTrack</tt> belongs to
+ * this TPC (that it has been previously added using {@link addTrack}). If the
+ * track does not belong an error message will be logged.
+ * @param {string} methodName the method name that will be logged in an error
+ * message
+ * @param {JitsiLocalTrack} localTrack
+ * @return {boolean} <tt>true</tt> if given local track belongs to this TPC or
+ * <tt>false</tt> otherwise.
+ * @private
+ */
+TraceablePeerConnection.prototype._assertTrackBelongs
+= function(methodName, localTrack) {
+    const doesBelong = this.localTracks.has(localTrack.rtcId);
+
+    if (!doesBelong) {
+        logger.error(
+            `${methodName}: ${localTrack} does not belong to ${this}`);
+    }
+
+    return doesBelong;
+};
+
+/**
+ * Checks whether given track is attached to this TPC. See
+ * {@link JitsiLocalTrack._isAttachedToPC} and {@link attachTrack} for more
+ * info.
+ * @param {JitsiLocalTrack} localTrack
+ * @return {boolean} <tt>true</tt> if attached or <tt>false</tt> otherwise
+ * @private
+ */
+TraceablePeerConnection.prototype._isTrackAttached = function(localTrack) {
```

Comment:
i thought the leading "_" signified that a method was intended to be 'private', and shouldn't be called from outside the class?